src/PFXImportPowershell/PFXImportPS/cmdlets/GetUserId.cs

```diff
@@ -71,6 +71,10 @@ public string UPN
         /// </summary>
         protected override void ProcessRecord()
         {
+            if(AuthenticationResult == null)
+            {
+                AuthenticationResult = (AuthenticationResult)SessionState.PSVariable.Get(Authenticate.AuthTokenKey)?.Value;
```

Comment:
If ```Get(Authenticate.AuthTokenKey)``` returns null, ```AuthenticationResult``` will still be ```null```. Does the next call (line 77), handle nulls?

Reply:
Yes